docs/diagrams/enhao/AddStudentActivityDiagram.puml

```diff
@@ -0,0 +1,27 @@
+@startuml
+start
+:User executes add student command;
+
+'Since the beta syntax does not support placing the condition outside the
+'diamond we place it as the true branch instead.
+
+if () then ([all prefixes are present])
+    : Parse all student's arguments;
+    if () then ([all arguments are valid])
+        :Logic executes the command;
+        if () then ([no duplicate student name])
+                : Model adds the student to the address book;
+                : Logic saves the address book to the storage;
+                : UI shows the updated student list;
```

Comment:
I think 'UI shows updated student list' comes before 'Logic saves the address book to the storage', according to the code:

```
commandResult = command.execute(model);
  try {
      storage.saveAddressBook(model.getAddressBook());
  } catch (IOException ioe) {
      throw new CommandException(FILE_OPS_ERROR_MESSAGE + ioe, ioe);
  }
```

You can double check!

Reply:
Ah ok! Seems logically, updated it already.